Functions.Rmd

```diff
@@ -469,23 +469,23 @@ x <- 20
 g12()
 ```
 
-This behaviour can be quite annoying. If you make a spelling mistake in your code, you won't get an error when you create the function, and you might not even get one when you run the function, depending on what variables are defined in the global environment. 
+This behaviour can be quite annoying. If you make a spelling mistake in your code, you won't get an error message when you create the function. And depending on the variables defined in the global environment, you might not even get an error message when you run the function. 
 
 \indexc{findGlobals()}
-One way to detect this problem is to use `codetools::findGlobals()`. This function lists all the external dependencies (unbound symbols) within a function: 
+One way to detect this problem is to use `codetools::findGlobals()`. This function lists all the external dependencies (unbound symbols) within a function:
 
 ```{r}
 codetools::findGlobals(g12)
 ```
 
-Another way to solve the problem would be to manually change the environment of the function to the `emptyenv()`, an environment which contains nothing:
+Another way to solve the problem is to manually change the function's environment to the `emptyenv()`, an environment which contains nothing:
 
 ```{r, error = TRUE}
 environment(g12) <- emptyenv()
 g12()
 ```
 
-Both of these approaches reveal why this undesirable behaviour exists: R relies on lexical scoping to find _everything_, even the `+` operator. This provides a rather beautiful simplicity to R's scoping rules.
+Both of these approaches reveal why this seemingly undesirable behaviour exists: R relies on lexical scoping to find _everything_, even the `+` operator. This gives R's scoping rules a rather beautiful simplicity.
```

Comment:
I think approaches should be examples?

Reply:
Re: "... R relies on lexical scoping to find _everything_, even the `+` operator"

To illustrate "_everything_", it might be nice to show a contrast. Something along the lines of: "from the humble `+` operator to [something less humble] ".  Suggestion for the latter?

Reply:
How about "from the obvious, like `mean()`, to the less obvious, like `+` and even `{`" ?